EIPS/eip-draft_micro_slots.md

```diff
@@ -0,0 +1,129 @@
+status: Draft
+type: Informational
+created: 2019-04-17
+---
+
+## Simple Summary
+The uint256 can store any value between 0 and 115792089237316195423570985008687907853269984665640564039457584007913129639935. 
+The uint256 data type is the most efficient and effective native Ethereum integer data type. 
+This informational EIP provides a standard of using modulo arithmetic and exponentiation to access and store multiple values inside a single uint256 slot.
+
+## Abstract
+There are many applications where a smart contract is required to store integer values which are much smaller than uint256. 
+Rather than declaring many smaller integers such as uint8 or bit arrays such as bytes1. Using the design pattern in this EIP, a smart contract could declare a single uint256 and then store and retrieve many more values than ordinarily possible.
+
+## Motivation
+bytes1 data type can only store 8 bits. Each bytes1 variable can therefore only store between 00000000 and 11111111 (0 to 255).
+If a smart contract has 32 bytes1 variables, it can store 32 individual values between 0 and 255; a total of 8160 (32 * 255 = 8160). However, if a smart contract implements this design patter, it could use a single uint256 variable (the equivalent in byte allocation i.e. a single bytes32 variable a.k.a. uint256) to store much more.
+
+## Specification
+The upper bound of uint256 is 
+```
+115792089237316195423570985008687907853269984665640564039457584007913129639935
+```
+
+Aside from the digit on the far left, there are 77 remaining digits which can have their value set to between zero and nine (0 to 9). One way to prove this is to execute the following Python3 script which returns the largest possible value of uint256 where all digits equal 9. Warning this test takes a considerable amount of time.
+
+```
+big = 115792089237316195423570985008687907853269984665640564039457584007913129639935
+
+
+for char in str(big):
+    if char != '9':
+        big = big - 1
+        break
+
+print (str(big))
+print (len(str(big)))
+```
+
+The results from the above test script are as follows
+
+```
+99999999999999999999999999999999999999999999999999999999999999999999999999999
+77
+```
+
+
+The factors of 77 are 1, 7, 11 and 77 and as such the following micro-slot combinations offer the most efficient storage with the least amount of waste.
+
+1, 77 digit integer with an individual value between 0 and 99999999999999999999999999999999999999999999999999999999999999999999999999999
+77, 1 digit integers with an individual value between 0 and 9
+7, 11 digit integers with an individual value between 0 and 99999999999
+11, 7 digit integers with an individual value between 0 and 9999999
+
+So how are the individual micro-slots accessed?
+
+### Function logic
+
+The following design pattern allows any digit[s] to be obtained from a single uint256 variable
+
+```
+((_value % (10 ** _position)) - (_value % (10 ** (_position - _size)))) / (10 ** (_position - _size));
```

Comment:
I am not entirely sure it is the most efficient way to split the 256-bit storage slot through the decimal representation with decimal arithmetic.

Bitwise shifting is available since Constantinople, which enables a highly efficient way to "split up" a 256-bit slot.